sympy/integrals/laplace.py

```diff
@@ -1089,6 +1096,208 @@ def simp_exp(arg):
     return _simplify(f.subs(t, t_), simplify), cond
 
 
+def _complete_the_square_in_denom(f, s):
+    from sympy.simplify.radsimp import fraction
+    [n, d] = fraction(f)
+    if d.is_polynomial(s):
+        cf = d.as_poly(s).all_coeffs()
+        if len(cf)==3:
+            a, b, c = cf
+            d = a*((s+b/(2*a))**2+c/a-(b/(2*a))**2)
+    return n/d
+
+
+@cacheit
+def _inverse_laplace_build_rules():
+    """
+    This is an internal helper function that returns the table of inverse
+    Laplace transform rules in terms of the time variable `t` and the
+    frequency variable `s`.  It is used by `_inverse_laplace_apply_rules`.
+    """
+    s = Dummy('s')
+    t = Dummy('t')
+    a = Wild('a', exclude=[s])
+    b = Wild('b', exclude=[s])
+    c = Wild('c', exclude=[s])
+    d = Wild('d', exclude=[s])
+
+    debug('_inverse_laplace_build_rules is building rules')
+    def _frac(f, s):
+        try:
+            return f.factor(s)
+        except PolynomialError:
+            return f
+    same = lambda f: f
+    frac = lambda f: _frac(f, s)
+    ctsd = lambda f: _complete_the_square_in_denom(f, s)
+    # This list is sorted according to the prep function needed.
+    _ILT_rules = [
+        (a/s, a, S.true, same, 1),
+        (b/(s+a), b*exp(-a*t), S.true, same, 1),
+        (a/s**2, a*t, S.true, same, 1),
+        (b/(s**2+a**2), b*sin(a*t)/a, S.true, same, 1),
+        (b/(s**2-a**2), b*sinh(a*t)/a, S.true, same, 1),
+        (b*s/(s**2+a**2), b*cos(a*t), S.true, same, 1),
+        (b*s/(s**2-a**2), b*cosh(a*t), S.true, same, 1),
+        (b*(s+a)**(-c), t**(c-1)*exp(-a*t)/gamma(c), c>0, same, 1),
+        #
+        (b/(s*(s+a)), b*(1-exp(-a*t))/a, S.true, frac, 1),
+        (b/(s+a)**2, b*t*exp(-a*t), S.true, frac, 1),
+        (b*s/(s+a)**2, b*(1-a*t)*exp(-a*t), S.true, frac, 1),
+        (c/((s+a)*(s+b)), c/(a-b)*(exp(-b*t)-exp(-a*t)), S.true, frac, 1),
+        (c*s/((s+a)*(s+b)), c/(a-b)*(a*exp(-a*t)-b*exp(-b*t)), S.true, frac, 1),
+        #
+        (c/(d*(s+b)**2+a**2),
+         c*exp(-b*t)*sin(a*t/sqrt(d))/(a*sqrt(d)), S.true, ctsd, 1),
+        (c/(d**2*(s+a)**2+b**2),
+         c/d*exp(-a*t)*(b*cos(b/d*t)/d-a*sin(b/d*t))/b,
+         S.true, ctsd, 1/s),
+    ]
+    return _ILT_rules, s, t
+
+
+def _inverse_laplace_apply_simple_rules(f, s, t):
+    """
+    Helper function for the class InverseLaplaceTransform.
+    """
+    if f==1:
+        debug('_inverse_laplace_apply_simple_rules match:')
+        debug('      f:    %s'%(1,))
+        debug('      rule: 1 o---o DiracDelta(%s)'%(t,))
+        return DiracDelta(t), S.true
+
+    _ILT_rules, s_, t_ = _inverse_laplace_build_rules()
+    _prep = ''
+
+    for s_dom, t_dom, check, prep, fac in _ILT_rules:
+        if not _prep is (prep, fac):
+            _F = prep(f.subs({s: s_})*fac)
```

Comment:
Try to do something like

```suggestion
    fsubs = f.subs({s: s_})
    _ILT_rules, s_, t_ = _inverse_laplace_build_rules()
    _prep = ''

    for s_dom, t_dom, check, prep, fac in _ILT_rules:
        if not _prep is (prep, fac):
            _F = prep(fsubs*fac)
```

and see what happens

Reply:
It works ;)